src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs

```diff
@@ -119,13 +119,11 @@ private async Task<ImmutableArray<SymbolReference>> DoAsync(SearchScope searchSc
                 searchScope.CancellationToken.ThrowIfCancellationRequested();
 
                 // Spin off tasks to do all our searching in parallel
-                var tasks = new List<Task<ImmutableArray<SymbolReference>>>
-                {
-                    GetReferencesForMatchingTypesAsync(searchScope),
-                    GetReferencesForMatchingNamespacesAsync(searchScope),
-                    GetReferencesForMatchingFieldsAndPropertiesAsync(searchScope),
-                    GetReferencesForMatchingExtensionMethodsAsync(searchScope),
-                };
+                using var _1 = ArrayBuilder<Task<ImmutableArray<SymbolReference>>>.GetInstance(out var tasks);
```

Comment:
📝 this one can just be a fixed-size local array allocation. The length is either 4 or 8 depending on `searchScope.Exact`.

Reply:
i'm ok just keeping the logic simple here :)